qa/tasks/cephfs/test_snap_schedules.py

```diff
@@ -445,4 +446,50 @@ def test_snap_schedule_with_mgr_restart(self):
         # cleanup
         self.fs_snap_schedule_cmd('remove', path=testdir, snap_schedule='1M')
         self.remove_snapshots(testdir[1:])
-        self.mount_a.run_shell(['rmdir', testdir[1:]])
+        self.mount_a.run_shell(['rmdir', testdir[1:]])    
+
+class TestSnapSchedulesSnapdir(TestSnapSchedulesHelper):
+    def remove_snapshots(self, dir_path, sdn):
+        snap_path = f'{dir_path}/{sdn}'
+
+        snapshots = self.mount_a.ls(path=snap_path)
+        for snapshot in snapshots:
+            snapshot_path = os.path.join(snap_path, snapshot)
+            log.debug(f'removing snapshot: {snapshot_path}')
+            self.mount_a.run_shell(['rmdir', snapshot_path])
+
+    def get_snap_dir_name(self):
+        from tasks.cephfs.fuse_mount import FuseMount
+        from tasks.cephfs.kernel_mount import KernelMount
+
+        if isinstance(self.mount_a, KernelMount):
+            sdn = self.mount_a.config.get('snapdirname', '.snap')
+        elif isinstance(self.mount_a, FuseMount):
+            sdn = self.mount_a.config.get('client_snapdir', '.snap')
```

Comment:
For `fuse_mount` its name should be `client_config`, while for `kernel_mount` it's name is `config`.

I think this should why it fail in pacific, if it still won't work then maybe you should backport https://github.com/ceph/ceph/pull/43981 and the following PRs for this.

Reply:
Please note the `kernel_mount` doesn't save the `config` as a local class member, such as the `fuse_mount` does by save it as `client_config` member.

Reply:
@lxbsz Isn't client_config a part of mount.py (CephFSMount)?

Reply:
@vshankar As I can check, client_config is part of CephFSMount only in main, not in pacific, quincy. And we didn't face this issue in quincy because https://github.com/ceph/ceph/pull/43981 exists in quincy and handled it but wasn't backported to pacific.

Reply:
Ah ok, the changes got picked up in quincy during branch out I guess. In that case, you can include that with this backport.

Reply:
Yeah. Correct.